OpenCL: GEMM kernel parameters error

Test log with #9988:
```
[ RUN      ] OCL_Core/Gemm.Accuracy/4
unknown file: Failure
C++ exception with description "/build/precommit_opencl_linux/opencv/modules/core/src/ocl.cpp:2266: error: (-220) OpenCL error CL_INVALID_ARG_INDEX (-49) during call: clSetKernelArg(arg_index=14, size=4, value=0x7ffdee028bf8) in function set
" thrown in the test body.
[  FAILED  ] OCL_Core/Gemm.Accuracy/4, where GetParam() = (32FC1, false, true, false, false) (1 ms)
[ RUN      ] OCL_Core/Gemm.Accuracy/5
unknown file: Failure
C++ exception with description "/build/precommit_opencl_linux/opencv/modules/core/src/ocl.cpp:2266: error: (-220) OpenCL error CL_INVALID_ARG_INDEX (-49) during call: clSetKernelArg(arg_index=14, size=4, value=0x7ffdee028bf8) in function set
" thrown in the test body.
[  FAILED  ] OCL_Core/Gemm.Accuracy/5, where GetParam() = (32FC1, false, true, false, true) (1 ms)
[ RUN      ] OCL_Core/Gemm.Accuracy/6
unknown file: Failure
C++ exception with description "/build/precommit_opencl_linux/opencv/modules/core/src/ocl.cpp:2266: error: (-220) OpenCL error CL_INVALID_ARG_INDEX (-49) during call: clSetKernelArg(arg_index=14, size=4, value=0x7ffdee028bf8) in function set
" thrown in the test body.
[  FAILED  ] OCL_Core/Gemm.Accuracy/6, where GetParam() = (32FC1, false, true, true, false) (1 ms)
[ RUN      ] OCL_Core/Gemm.Accuracy/7
unknown file: Failure
C++ exception with description "/build/precommit_opencl_linux/opencv/modules/core/src/ocl.cpp:2266: error: (-220) OpenCL error CL_INVALID_ARG_INDEX (-49) during call: clSetKernelArg(arg_index=14, size=4, value=0x7ffdee028bf8) in function set
" thrown in the test body.
[  FAILED  ] OCL_Core/Gemm.Accuracy/7, where GetParam() = (32FC1, false, true, true, true) (2 ms)
```

Similar logs in perf_core tests.

--- a/modules/core/src/intel_gpu_gemm.inl.hpp
+++ b/modules/core/src/intel_gpu_gemm.inl.hpp
@@ -108,9 +108,8 @@ static bool intel_gpu_gemm(
                (float)beta,
                (int)(A.step / sizeof(float)),
                (int)(B.step / sizeof(float)),
-               (int)(D.step / sizeof(float)),
-               (int) 0,                          // 14 start_index
-               stride);
+               (int)(D.step / sizeof(float))
+        );
 
         ret = k.run(2, global, local, false, q);
     }

